spectral_cube/spectral_cube.py

```diff
@@ -2287,6 +2299,363 @@ def find_lines(self, velocity_offset=None, velocity_convention=None,
+                                   wcs=wcs_utils.slice_wcs(self._wcs, view),
+                                   mask=newmask,
+                                   beams=self.beams[specslice],
+                                   meta=meta)
+
+    def spectral_slab(self, lo, hi):
+        """
+        Extract a new cube between two spectral coordinates
+
+        Parameters
+        ----------
+        lo, hi : :class:`~astropy.units.Quantity`
+            The lower and upper spectral coordinate for the slab range. The
+            units should be compatible with the units of the spectral axis.
+            If the spectral axis is in frequency-equivalent units and you
+            want to select a range in velocity, or vice-versa, you should
+            first use :meth:`~spectral_cube.SpectralCube.with_spectral_unit`
+            to convert the units of the spectral axis.
+        """
+
+        # Find range of values for spectral axis
+        ilo = self.closest_spectral_channel(lo)
+        ihi = self.closest_spectral_channel(hi)
+
+        if ilo > ihi:
+            ilo, ihi = ihi, ilo
+        ihi += 1
+
+        # Create WCS slab
+        wcs_slab = self._wcs.deepcopy()
+        wcs_slab.wcs.crpix[2] -= ilo
+
+        # Create mask slab
+        if self._mask is None:
+            mask_slab = None
+        else:
+            try:
+                mask_slab = self._mask[ilo:ihi, :, :]
+            except NotImplementedError:
+                warnings.warn("Mask slicing not implemented for "
+                              "{0} - dropping mask".
+                              format(self._mask.__class__.__name__))
+                mask_slab = None
+
+        # Create new spectral cube
+        slab = self._new_cube_with(data=self._data[ilo:ihi], wcs=wcs_slab,
+                                   beams=self.beams[ilo:ihi],
+                                   mask=mask_slab)
+
+        return slab
+
+    def _new_cube_with(self, **kwargs):
+        beams = kwargs.pop('beams', self.beams)
+        beam_threshold = kwargs.pop('beam_threshold', self.beam_threshold)
+        VRSC = VaryingResolutionSpectralCube
+        newcube = super(VRSC, self)._new_cube_with(beams=beams,
+                                                   beam_threshold=beam_threshold,
+                                                   **kwargs)
+        return newcube
+
+    _new_cube_with.__doc__ = SpectralCube._new_cube_with.__doc__
+
+    def _check_beam_areas(self, threshold):
+        """
+        Check that the beam areas are the same to within some threshold
+        """
+
+        qtys = dict(area = u.Quantity(self.beams, u.sr),
+                    major = u.Quantity([bm.major for bm in self.beams], u.deg),
+                    minor = u.Quantity([bm.minor for bm in self.beams], u.deg),
+                    pa = u.Quantity([bm.pa for bm in self.beams], u.deg),
+                   )
+
+        errormessage = ""
+
+        for qtyname, qty in qtys.items():
```

Comment:
`cube_utils.average_beams` should be used to get the mean values (for consistency).

For the PA, I guess the comparison should be with the max absolute difference from the mean?